reference-implementation/lib/writable-stream.js

```diff
@@ -188,7 +188,14 @@ function WritableStreamError(stream, e) {
 }
 
 function WritableStreamFinishClose(stream) {
-  assert(stream._state === 'closing' || stream._state === 'errored');
+  assert(stream._pendingCloseRequest !== undefined);
+  stream._pendingCloseRequest._resolve(undefined);
+  stream._pendingCloseRequest = undefined;
+
+  if (stream._pendingAbortRequest !== undefined) {
```

Comment:
Oh, it's from L202 of the previous commit. I changed the order to make this and `WritableStreamFinishCloseWithError()` look similar. This reordering shouldn't cause any behavior change.